### PR TITLE
Trenches - Improve by using surfaceTexture

### PR DESCRIPTION
**Pull request by Cyruz143**
**When merged this pull request will:**
- Make use of `surfaceTexture` for trenches
- Just a direct port of https://github.com/acemod/ACE3/pull/7833 without hardcoded textures

Hopefully it's ok with @Mysteryjuju as it's just a copy/paste of their PR but switched to use surfaceTexture.

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

**Comment by Drofseh** [comment]
I strongly disagree with have the trench immediately get the ground texture like this. A freshly dug trench will be the colour of the dirt it's dug in.
It should be an action on the trench to camouflage it or something like that to give it the ground texture.

**Comment by Cyruz143** [comment]
I agree that would be nice, however in it's current implementation it's pretty much always just green grass which looks fairly out of place on most terrains, especially snow/desert ones etc. This is more a bandaid to make it look a bit less shit until someone wants to go through and revamp them all as per the other failed trenches PR.

**Comment by Drofseh** [comment]
They currently are placed as dirt though, which I think is correct.

Should be simple to just add an interaction to either the objects or class that then applies the texture.

**Comment by Cyruz143** [comment]
They look like this I believe:
https://i.imgur.com/9PX3Sjw_d.webp?maxwidth=640&shape=thumb&fidelity=medium

I took the image from Google as not near a computer currently.

**Comment by Drofseh** [comment]
Ahh yeah the current texture is not good either, but I think having there be a better default texture at the start is completely fine, I just don't want it to be set to the surfaceTexture automatically

I think any one of these would work as the default texture, though I'm partital to mud.
```
"a3\map_data\gdt_marsh_co.paa"
"a3\map_data\gdt_mud_co.paa"
"a3\map_data\gdt_soil_co.paa"
```
351ffb649f75346d99ec56011b6720ed

**Comment by Cyruz143** [comment]
Added feedback. Not sure if we want another progress bar/setting for time to apply camo? To be honest I'm not a fan of endless bars and more faffing around.

File: addons/trenches/stringtable.xml [hunk 1]
```diff
@@ -226,6 +226,22 @@
             <Turkish>Siperi Kaldır</Turkish>
             <Spanish>Eliminar trinchera</Spanish>
         </Key>
+        <Key ID="STR_ACE_Trenches_CamouflageTrench">
+            <English>Camouflage Trench</English>
+            <Polish></Polish>
```
Comment (veteran29) [inline addons/trenches/stringtable.xml:231]:
```suggestion
            <Polish>Zakamufluj Okop</Polish>
```

Empty translations should be removed.

**Comment by veteran29** [comment]
> Added feedback. Not sure if we want another progress bar/setting for time to apply camo? To be honest I'm not a fan of endless bars and more faffing around.

Maybe `_player playActionNow "PutDown";` for minimal feedback and that's it?

File: addons/trenches/functions/fnc_camouflageTrench.sqf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include "script_component.hpp"
+/*
+ * Author: Cyruz
+ * Apply surfaceTexture to a completed trench
```
Comment (jonpas) [inline addons/trenches/functions/fnc_camouflageTrench.sqf:4]:
```suggestion
 * Apply surfaceTexture to a completed trench.
```

File: addons/trenches/functions/fnc_camouflageTrench.sqf [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Arguments:
+ * 0: trench <OBJECT>
```
Comment (jonpas) [inline addons/trenches/functions/fnc_camouflageTrench.sqf:7]:
```suggestion
 * 0: Trench <OBJECT>
```

File: addons/trenches/functions/fnc_camouflageTrench.sqf [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ * 1: unit <OBJECT>
```
Comment (jonpas) [inline addons/trenches/functions/fnc_camouflageTrench.sqf:8]:
```suggestion
 * 1: Unit <OBJECT>
```

File: addons/trenches/functions/fnc_camouflageTrench.sqf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+params ["_trench", "_unit"];
+
+_unit playActionNow "PutDown";
+
+[{
+    _this setObjectTextureGlobal [0, (surfaceTexture getPosATL _this)];
+    _this setVariable [QGVAR(camouflaged), true, true];
+}, _trench, 2] call CBA_fnc_waitAndExecute;
+
```
Comment (jonpas) [inline addons/trenches/functions/fnc_camouflageTrench.sqf:27]:
```suggestion
```

File: addons/trenches/functions/fnc_camouflageTrench.sqf [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * Public: No
+ */
+
+params ["_trench", "_unit"];
+
+_unit playActionNow "PutDown";
+
+[{
+    _this setObjectTextureGlobal [0, (surfaceTexture getPosATL _this)];
```
Comment (jonpas) [inline addons/trenches/functions/fnc_camouflageTrench.sqf:24]:
```suggestion
    _this setObjectTextureGlobal [0, surfaceTexture getPosATL _this];
```

File: addons/trenches/functions/fnc_canCamouflageTrench.sqf [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: Cyruz
+ * Checks if a unit can camouflage a trench
```
Comment (jonpas) [inline addons/trenches/functions/fnc_canCamouflageTrench.sqf:4]:
```suggestion
 * Checks if a unit can camouflage a trench.
```

File: addons/trenches/functions/fnc_canCamouflageTrench.sqf [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * Arguments:
+ * 0: trench <OBJECT>
+ * 1: unit <OBJECT>
```
Comment (jonpas) [inline addons/trenches/functions/fnc_canCamouflageTrench.sqf:8]:
```suggestion
 * 1: Unit <OBJECT>
```

File: addons/trenches/functions/fnc_canCamouflageTrench.sqf [hunk 9]
```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: Cyruz
+ * Checks if a unit can camouflage a trench
+ *
+ * Arguments:
+ * 0: trench <OBJECT>
```
Comment (jonpas) [inline addons/trenches/functions/fnc_canCamouflageTrench.sqf:7]:
```suggestion
 * 0: Trench <OBJECT>
```

**Review by jonpas — DISMISSED** [review]

**Review by kymckay — APPROVED** [review]
Nice. There's a small edge case where the player could be killed during the 2s delay to camouflage (so it would become camouflaged while they're unconscious/dead), but I don't think that's meaningful enough to be worth explicitly handling.

**Comment by Cyruz143** [comment]
> Nice. There's a small edge case where the player could be killed during the 2s delay to camouflage (so it would become camouflaged while they're unconscious/dead), but I don't think that's meaningful enough to be worth explicitly handling.

I did think about it but it's a minor chance and even if it gets camouflaged as they're uncon/dead, it's not the end of the world.

**Comment by Cyruz143** [comment]
Can someone close off https://github.com/acemod/ACE3/pull/7833 as this has been merged.

**Pull request merged**